app/pages/project/workflow-selection.jsx

```diff
@@ -23,8 +23,9 @@ class WorkflowSelection extends React.Component {
       nextProps.preferences.preferences.selected_workflow &&
       this.state.workflow
     ) {
+      const userWorkflowID = parseInt(nextProps.preferences.preferences.selected_workflow);
       if (!nextState.loadingSelectedWorkflow &&
-        nextProps.preferences.preferences.selected_workflow !== this.state.workflow.id
```

Comment:
Is this `parseInt` really necessary? Shouldn't it come back as an int already, since JSON has data types?

Reply:
When I reproduced the bug, the stored workflow ID was coming back from the preferences as `6605env=production` so I think this is definitely needed.

Reply:
No I meant on the `parseInt(this.state.workflow.id)`. Isn't `state.workflow` filled with an object coming from the Panoptes API?

Reply:
I think that compares `6605 !== ‘6605’`. Resource IDs are strings, for some reason.

Reply:
I've tested this now. If I take out that `parseInt`, the exact comparison continuously fails and the component spams the API with requests.

Reply:
Also, looking at the network responses, resource IDs are sent back as strings from Panoptes.